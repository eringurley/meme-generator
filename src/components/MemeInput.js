import React from 'react';
import PropTypes from 'prop-types';

function MemeInput({ header, imgURL, footer, handleChange }) {
  return (
    <>
      <form>
        <input name="header" value={header} placeholder="header text" onChange={(event) => handleChange(event)}></input>
        <input name="imgURL" value={imgURL} placeholder="imgURL text" onChange={(event) => handleChange(event)}></input>
        <input name="footer" value={footer} placeholder="footer text" onChange={(event) => handleChange(event)}></input>
        <button>Meme Me</button>
      </form>
    </>
  );
}

MemeInput.propTypes = {
  header: PropTypes.string.isRequired,
  imgURL: PropTypes.string.isRequired,
  footer: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired
};

export default MemeInput;
