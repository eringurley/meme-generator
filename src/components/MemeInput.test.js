import React from 'react';
import { shallow } from 'enzyme';
import MemeInput from './MemeInput';

describe('MemeInput component', () => {
  it('renders MemeInput', () => {
    const handleChange = jest.fn();
    const wrapper = shallow(<MemeInput header="hi" imgURL="image" footer="footer" handleChange={handleChange}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
