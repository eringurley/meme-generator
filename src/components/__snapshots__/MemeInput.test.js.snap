// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemeInput component renders MemeInput 1`] = `
<Fragment>
  <form>
    <input
      name="header"
      onChange={[Function]}
      placeholder="header text"
      value="hi"
    />
    <input
      name="imgURL"
      onChange={[Function]}
      placeholder="imgURL text"
      value="image"
    />
    <input
      name="footer"
      onChange={[Function]}
      placeholder="footer text"
      value="footer"
    />
  </form>
</Fragment>
`;
